pkg/kubelet/dockershim/security_context.go

```diff
@@ -68,16 +76,35 @@ func applyContainerSecurityContext(lc *runtimeapi.LinuxContainerConfig, podSandb
 }
 
 // modifyContainerConfig applies container security context config to dockercontainer.Config.
-func modifyContainerConfig(sc *runtimeapi.LinuxContainerSecurityContext, config *dockercontainer.Config) {
+func modifyContainerConfig(sc *runtimeapi.LinuxContainerSecurityContext, config *dockercontainer.Config) error {
 	if sc == nil {
-		return
+		return fmt.Errorf("LinuxContainerSecurityContext is nil")
 	}
 	if sc.RunAsUser != nil {
 		config.User = strconv.FormatInt(sc.GetRunAsUser().Value, 10)
 	}
 	if sc.RunAsUsername != "" {
 		config.User = sc.RunAsUsername
 	}
+
+	user := config.User
+	if sc.RunAsGroup != nil {
+		if user == "" {
+			return fmt.Errorf("Group is specified without a User, %s", sc.RunAsGroup)
+		}
+		gid := strconv.FormatInt(sc.GetRunAsGroup().Value, 10)
+		user = fmt.Sprintf("%s:%s", config.User, gid)
+	}
+
+	if sc.RunAsGroupname != "" {
+		if user == "" {
+			return fmt.Errorf("Group is specified without a User, %s", sc.RunAsGroupname)
+		}
+		user = fmt.Sprintf("%s:%s", config.User, sc.RunAsGroupname)
+	}
+	config.User = user
+
+	return nil
```

Comment:
@pmorie @liggitt can you guys review this section which is the meat of the implementation.